core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -85,13 +87,50 @@
   @Nullable
   private SubchannelPicker lastPicker;
 
-  DelayedClientTransport2(Executor streamCreationExecutor) {
-    this.streamCreationExecutor = streamCreationExecutor;
+  @GuardedBy("lock")
+  private long lastPickerVersion;
+
+  /**
+   * Creates a new delayed transport.
+   *
+   * @param defaultAppExecutor pending streams will create real streams and run bufferred operations
+   *        in an application executor, which will be this executor, unless there is on provided in
+   *        {@link CallOptions}.
+   * @param channelExecutor all listener callbacks of the delayed transport will be run from this
```

Comment:
Delayed transport's `transportInUse()` callbacks are now serialized with the rest of the channel impl logic, including Subchannel's in-use callbacks and the idle timer's actual logic, making the idleness logic totally serialized thus much simpler.